tests/posix/common/src/nanosleep.c

```diff
@@ -126,15 +126,23 @@ static void common_lower_bound_check(int selection, clockid_t clock_id, int flag
 	int r;
 	uint64_t actual_ns;
 	uint64_t exp_ns;
-	uint32_t now;
-	uint32_t then;
+	uint64_t now;
+	uint64_t then;
 	struct timespec rem = {0, 0};
 	struct timespec req = {s, ns};
 
 	errno = 0;
+#if CONFIG_TIMER_HAS_64BIT_CYCLE_COUNTER
+	then = k_cycle_get_64();
+#else
 	then = k_cycle_get_32();
+#endif
 	r = select_nanosleep(selection, clock_id, flags, &req, &rem);
+#if CONFIG_TIMER_HAS_64BIT_CYCLE_COUNTER
+	now = k_cycle_get_64();
+#else
 	now = k_cycle_get_32();
+#endif
```

Comment:
@cfriedt , the ksleep is a usleep, which ROUND_UP the ns sleep time, so there will be some accuracy problem in our platform when the sleep time is very short, so I change it to k_cycle_get_64, which will get more pricise clock, however, Nordic simulation platform does not support k_cycle_get_64 so I add this code, this is to resolve the problem introduced by this `+then` fixes.

Reply:
Great - can you love it to a separate commit please? I'll add some additional change requests for that code now so that you don't need to respin later.

Reply:
done